packages/atomic/src/utils/dom-utils.spec.ts

```diff
@@ -0,0 +1,269 @@
+      const rect2 = new DOMRect(-10, -20, 100, 200);
+
+      expect(rectEquals(rect1, rect2)).toBe(true);
+    });
+
+    it('should handle floating point values correctly', () => {
+      const rect1 = new DOMRect(10.5, 20.7, 100.3, 200.9);
+      const rect2 = new DOMRect(10.5, 20.7, 100.3, 200.9);
+
+      expect(rectEquals(rect1, rect2)).toBe(true);
+    });
+  });
+
+  describe('#parentNodeToString', () => {
+    it('should return empty string for node with no children', () => {
+      const emptyDiv = document.createElement('div');
+
+      expect(parentNodeToString(emptyDiv)).toBe('');
+    });
+
+    it('should return HTML string for node with single child', () => {
+      const parent = document.createElement('div');
+      const child = document.createElement('span');
+      child.textContent = 'Hello';
+      parent.appendChild(child);
+
+      expect(parentNodeToString(parent)).toBe('<span>Hello</span>');
+    });
+
+    it('should return concatenated HTML for node with multiple children', () => {
+      const parent = document.createElement('div');
+
+      const child1 = document.createElement('span');
+      child1.textContent = 'First';
+
+      const child2 = document.createElement('p');
+      child2.textContent = 'Second';
+
+      parent.appendChild(child1);
+      parent.appendChild(child2);
+
+      expect(parentNodeToString(parent)).toBe(
+        '<span>First</span><p>Second</p>'
+      );
+    });
+
+    it('should handle children with attributes', () => {
+      const parent = document.createElement('div');
+      const child = document.createElement('a');
+      child.href = 'https://example.com';
+      child.className = 'link';
+      child.textContent = 'Link';
+      parent.appendChild(child);
+
+      expect(parentNodeToString(parent)).toBe(
+        '<a href="https://example.com" class="link">Link</a>'
+      );
+    });
+
+    it('should handle nested children', () => {
+      const parent = document.createElement('div');
+      const child = document.createElement('div');
+      const grandchild = document.createElement('span');
+      grandchild.textContent = 'Nested';
+      child.appendChild(grandchild);
+      parent.appendChild(child);
+
+      expect(parentNodeToString(parent)).toBe('<div><span>Nested</span></div>');
+    });
+
+    it('should handle document fragment', () => {
+      const fragment = document.createDocumentFragment();
+      const child1 = document.createElement('span');
+      child1.textContent = 'Fragment child 1';
+      const child2 = document.createElement('div');
+      child2.textContent = 'Fragment child 2';
+
+      fragment.appendChild(child1);
+      fragment.appendChild(child2);
+
+      expect(parentNodeToString(fragment)).toBe(
+        '<span>Fragment child 1</span><div>Fragment child 2</div>'
+      );
+    });
+  });
+
+  describe('#closest', () => {
+    it('should return the element itself if it matches the selector', () => {
```

Comment:
```suggestion
    it('should return the element itself when it matches the selector', () => {
```

Reply:
What happens when both the element itself and, say, its parent match the selector?

I guess it would return the element itself, but that'd be a good test I think.